streams/dshr_stream_mod.F90

```diff
@@ -1964,28 +2017,35 @@ subroutine shr_stream_restIO(pioid, streams, mode)
        rcode = pio_inq_varid(pioid, 'timeofday', tvarid)
        rcode = pio_inq_varid(pioid, 'haveData' , hdvarid)
        do k=1,size(streams)
-          logunit = streams(k)%logunit
           do n=1,streams(k)%nfiles
 
              ! read in filename
              rcode = pio_get_var(pioid, varid, (/1,n,k/), fname)
-             
+
              if(trim(fname) /= trim(streams(k)%file(n)%name)) then
-                write(logunit,*) 'Filename does not match restart record, checking realpath'
+                if (mainproc) then
+                   write(logout,'(2a)') trim(subname),'Filename does not match restart record, checking realpath'
+                end if
                 call shr_file_get_real_path(fname, rfname)
                 call shr_file_get_real_path(trim(streams(k)%file(n)%name), rsfname)
                 if (trim(rfname) /= trim(rsfname)) then
-                   write(logunit,*) 'Filename path does not match restartfile, checking filename'
+                   if (mainproc) then
+                      write(logout,'(2a)') trim(subname),'Filename path does not match restartfile, checking filename'
+                   end if
                    rfname = fname(index(fname,'/',.true.):)
                    rsfname = streams(k)%file(n)%name(index(streams(k)%file(n)%name, '/',.true.):)
                    if (trim(rfname) /= trim(rsfname)) then
-                      write(logunit,*) trim(rfname), '<>', trim(rsfname)
-                      write(logunit,'(a)')' fname = '//trim(fname)
-                      write(logunit,'(a,i8,2x,i8,2x,a)')' k,n,streams(k)%file(n)%name = ',k,n,trim(streams(k)%file(n)%name)
+                      if (mainproc) then
+                         write(logout,'(2a)') trim(subname),trim(rfname), '<>', trim(rsfname)
```

Comment:
I see four fields:
```suggestion
                         write(logout,'(4a)') trim(subname),trim(rfname), '<>', trim(rsfname)
```
Also, `subname` is a parameter (here and elsewhere in this subroutine) and therefore, does not need to be `trim`med.